test/e2e/conditions.go

```diff
@@ -73,3 +73,40 @@ func enactmentConditionsStatusEventually(node string) AsyncAssertion {
 func enactmentConditionsStatusConsistently(node string) AsyncAssertion {
 	return enactmentConditionsStatusForPolicyConsistently(node, TestPolicy)
 }
+
+func policyConditionsStatus(policyName string) nmstatev1alpha1.ConditionList {
+	policy := nodeNetworkConfigurationPolicy(types.NamespacedName{Name: policyName})
+	obtainedConditions := nmstatev1alpha1.ConditionList{}
+	for _, policysConditionType := range nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionTypes {
+		obtainedCondition := policy.Status.Conditions.Find(policysConditionType)
+		obtainedConditionStatus := corev1.ConditionUnknown
+		if obtainedCondition != nil {
+			obtainedConditionStatus = obtainedCondition.Status
+		}
+		obtainedConditions = append(obtainedConditions, nmstatev1alpha1.Condition{
+			Type:   policysConditionType,
+			Status: obtainedConditionStatus,
+		})
+	}
+	return obtainedConditions
+}
+
+func policyConditionsStatusForPolicyEventually(policy string) AsyncAssertion {
+	return Eventually(func() nmstatev1alpha1.ConditionList {
+		return policyConditionsStatus(policy)
+	}, 180*time.Second, 1*time.Second)
+}
+
+func policyConditionsStatusForPolicyConsistently(policy string) AsyncAssertion {
```

Comment:
Because golagn does not support function override, we have two functions one that uses "test-policy" and the other that specify the policy you want to use.

Reply:
I was just proposing change of the naming